test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -4334,6 +4476,274 @@ def test_cgroup_default_config(self):
+                             % str(cgswap))
+        else:
+            self.logger.info('First MoM has swap, confirming cgswap '
+                             'correctly throttles jobs accepted')
+            # PbsTypeSize value is stored in kb units
+            cgreqval = int(float(cgswap.value)
+                           / 1024.0 / 3.0 * 2.0)
+            cgreqsuffix = 'mb'
+            cgreq = PbsTypeSize(str(cgreqval) + cgreqsuffix)
+            vmemreqsize = PbsTypeSize("100mb") + cgreq
+            vmemreq = str(int(vmemreqsize.value / 1024))+'mb'
+            self.logger.info('will submit jobs with 100mb mem and %s vmem'
+                             % vmemreq)
+            a = {'Resource_List.select':
+                 '1:ncpus=1:mem=100mb:vmem=%s:vnode=%s'
+                 % (vmemreq, vnode_name)}
+
+            j = Job(TEST_USER, attrs=a)
+            j.create_script(self.sleep100_job)
+            jid = self.server.submit(j)
+            bs = {'job_state': 'R'}
+            self.server.expect(JOB, bs, jid, offset=1)
+
+            cgswap = None
+            job_status = self.server.status(JOB, id=jid)
+            select_resource = job_status[0]['Resource_List.select']
+            chunkspecs = select_resource.split(':')
+            for c in chunkspecs:
+                if '=' in c:
+                    name, value = c.split('=')
+                    if name == 'cgswap':
+                        cgswap = PbsTypeSize(value)
+            self.assertTrue(cgswap is not None, 'job cgswap was not added')
+            self.assertTrue(cgswap == cgreq,
+                            'job cgswap is %s instead of expected %s'
+                            % (str(cgswap), str(cgreq)))
+            self.logger.info('job cgswap detected to be correct, roughly %s'
+                             % str(cgswap))
+            j = Job(TEST_USER, attrs=a)
+            j.create_script(self.sleep100_job)
+            jid = self.server.submit(j)
+
+            # Second job should not run - not enough cgswap
+            # offset large enough to give scheduler a chance
+            bs2 = {'job_state': 'Q'}
+            self.server.expect(JOB, bs2, jid, offset=20, max_attempts=1)
+
+    def test_cgroup_cgswap_numa(self):
+        self.test_cgroup_cgswap(vnode_per_numa_node=True)
+
+    def test_cgroup_enforce_default(self,
+                                    enforce_flags=('true', 'true'),
+                                    exclhost=False):
+        if not self.mem:
+            self.skipTest('Test requires memory subystem mounted')
+        if not self.swapctl:
+            self.skipTest('Test requires memsw accounting enabled')
+
+        self.load_config(self.cfg16
+                         % enforce_flags)
+        # Restart mom for changes made by cgroups hook to take effect
+        begin = time.time()
+        # sleep 2s to ensure log matching will not catch older log lines
+        time.sleep(2)
```

Comment:
Same comment as earlier, can we get rid of the 2 second sleep?

Reply:
I would suggest not to do so, but I will rewrite it as before.